src/app/lib/definitions.ts

```diff
@@ -1,11 +1,38 @@
 export type Settings = {
-    baseURL: string
+  baseURL: string
 }
 
 
-export type Step = {
-    message: string
-    deadline: number
-    stage?: string
-    comment?: string
+export type BrewingStep = {
+  message: string
+  deadline: number
+  stage?: string
+  comment?: string
+}
+
+export type BrewingRecipe = {
+  author: string
+  name: string
+  source: string
+  description: string
+  coffeeAmount: number
+  waterAmount: number
+  brewingSteps: Array<BrewingStep>
+}
+
+export type RecipeStep = {
+  messageTemplate: string
+  duration: number
+  amount?: number
+  stage?: string
+  commentTemplate?: string
+}
+
+export type Recipe = {
```

Comment:
Could you add a comment (or rename)? I'm not sure I understand what the difference is between `BrewingRecipe` and `Recipe`.

`Reciepe` is a raw reciepie from JSON, isn't it?

Reply:
I'm thinking about RawReciepe or RawReciepeSchema (the instance is usually called like that with `raw`) but if you don't want to change it, it's ok